src/main/java/org/jetbrains/research/refactorinsight/ui/DiffHintLineMarkerProvider.java

```diff
@@ -69,24 +63,52 @@ public void collectSlowLineMarkers(@NotNull List<? extends PsiElement> elements,
         if (refactoringInfos == null) return;
 
         boolean isRight = isRightPartOfDiff(elements.get(0));
-        Map<Integer, Set<RefactoringInfo>> refactoringsMap = new HashMap<>();
+        Map<Integer, List<RefactoringWithPsiElement>> lineToRefactorings = new HashMap<>();
```

Comment:
maybe it could be rewritten this way?
```java
        Map<Integer, Set<RefactoringWithPsiElement>> lineToRefactorings = new HashMap<>();

        for (PsiElement element : elements) {
            if (!isIdentifier(element)) continue;
            int lineNumber = getLineNumber(element);
            int textOffset = element.getTextOffset();
            for (RefactoringInfo refactoringInfo : refactoringInfos) {
                if (refactoringInfo.containsElement(lineNumber, textOffset, isRight)) {
                    RefactoringWithPsiElement refactoringWithPsiElement = new RefactoringWithPsiElement(refactoringInfo, element);
                    lineToRefactorings.computeIfAbsent(lineNumber, __ -> new LinkedHashSet<>())
                            .add(refactoringWithPsiElement);
                }
            }
        }
        
        for (Set<RefactoringWithPsiElement> value : lineToRefactorings.values()) {
            var psiElements = value.stream().map(it -> it.element).collect(Collectors.toSet());
            var refactorings = value.stream().map(it -> it.info).toList();
            result.add(new RefactoringInfoHint(new SmartList<>(psiElements), refactorings));
        }
```